lisa/sut_orchestrator/azure/arm_template.bicep

```diff
@@ -0,0 +1,323 @@
+}
+
+func getEphemeralOSImage(node object) object => {
+  name: '${node.name}-osDisk'
+  diffDiskSettings: {
+      option: 'local'
+      placement: 'CacheDisk'
+  }
+  caching: 'ReadOnly'
+  createOption: 'FromImage'
+  diskSizeGB: node.osdisk_size_in_gb
+}
+
+func getAttachDisk(disk object, diskName string, index int) object => {
+  lun: index
+  createOption: 'attach'
+  caching: disk.caching_type
+  managedDisk: {
+      id: resourceId('Microsoft.Compute/disks', diskName)
+  }
+}
+
+func getOsDiskSharedGallery(node object) object => {
+  id: resourceId(node.subscription_id, empty(node.resource_group_name) ? 'None' : node.resource_group_name, 'Microsoft.Compute/galleries/images/versions', node.image_gallery, node.image_definition, node.image_version)
+}
+
+func getOsDiskMarketplace(node object) object => {
+  publisher: node.marketplace.publisher
+  offer: node.marketplace.offer
+  sku: node.marketplace.sku
+  version: node.marketplace.version
+}
+
+func getSPWithDiskEncryptionSet(node object) object => {
+  securityEncryptionType: node.security_profile.encryption_type
+  diskEncryptionSet: {
+    id: node.security_profile.disk_encryption_set_id
+  }
+}
+
+func getSP(node object) object => {
+  securityEncryptionType: node.security_profile.encryption_type
+}
+
+func getOSImage(node object) object => {
+  name: '${node.name}-osDisk'
+  managedDisk: {
+    storageAccountType: node.os_disk_type
+    securityProfile: ((empty(node.security_profile) || (node.security_profile.security_type != 'ConfidentialVM')) ? null : (empty(node.security_profile.disk_encryption_set_id) ? getSP(node) : getSPWithDiskEncryptionSet(node)))
+  } 
+  caching: ((node.os_disk_type == 'Ephemeral') ? 'ReadOnly' : 'ReadWrite')
+  createOption: 'FromImage'
+  diskSizeGB: node.osdisk_size_in_gb
+}
+
+func getSecurityProfile(security_profile object) object => {
+  uefiSettings: {
+    secureBootEnabled: security_profile.secure_boot
+    vTpmEnabled: true
+  }
+  securityType: security_profile.security_type
+}
+
+func getAvailabilitySetId(availability_set_name string) object => {
+  id: resourceId('Microsoft.Compute/availabilitySets', availability_set_name)
+}
+
+func hasVmgsPath(node object) bool => bool((!empty(node.vhd)) && (!empty(node.vhd.vmgs_path)))
+
+func hasVhdPath(node object) bool => bool((!empty(node.vhd)) && (!empty(node.vhd.vhd_path)))
+
+resource nodes_vms 'Microsoft.Compute/virtualMachines@2022-08-01' = [for i in range(0, node_count): {
+  name: nodes[i].name
+  location: nodes[i].location
+  tags: vm_tags
+  plan: nodes[i].purchase_plan
+  properties: {
+    availabilitySet: availability_set_value
+    hardwareProfile: {
+      vmSize: nodes[i].vm_size
+    }
+    osProfile: (hasVmgsPath(nodes[i]) ? null : getOsProfile(nodes[i].short_name, admin_username, (!empty(admin_password)), admin_password, ((!empty(admin_key_data)) && nodes[i].is_linux), getLinuxConfiguration('/home/${admin_username}/.ssh/authorized_keys', admin_key_data, empty(admin_password))))
```

Comment:
The `hasVmgsPath` should be called in `getOsProfile`.

Reply:
I tried to wrap them in one function, please take a look, thanks